r/tools/nixlibs.R

```diff
@@ -192,17 +192,12 @@ download_binary <- function(lib) {
 # of action based on the current system. Other values you can set it to:
 # * "FALSE" (not case-sensitive), to skip this option altogether
 # * "TRUE" (not case-sensitive), to try to discover your current OS, or
-# * Some other string: a "linux-openssl-${OPENSSL_VERSION}" that corresponds to
-#   a binary that is available, to override what this function may discover by
-#   default.
+# * Some other string: a binary identifier that corresponds to a binary that is
```

Comment:
Since it _is_ possible for someone to pass these binary names in, do we need to be concerned with backwards compatibility for user code that say has `LIBARROW_BINARY=linux-x86_64-openssl-3.0`?

If we wanted to handle that, I think the logic would be:

* grep for `openssl-1`, error if found
* optionally, grep for `openssl-3` and print a deprecation warning
* then `sub("-openssl-3\\.0$", "", x)` (drop the suffix)

Reply:
Updated